File: BOS_correlation_OpenPIV.m
function [Displ] = BOS_correlation_OpenPIV(im1,im2,nx,ny,overlap_x)
% cross-correlation of two images, using OpenPIV (www.openpiv.net)
% Inputs:
%   im1,im2 - images
% Outputs:
%   Displ - displacement field, dx,dy

addpath('./openpiv');


% Overlap in pixels
overlap_px = nx*overlap_x;   % pix
overlap_py = nx*overlap_x;   % pix


% Compute the cross-correlation using OpenPIV
% Note that openpiv also saves .VEC file in the image folder if you need
% it later
% loadvec([imfile1,'.vec']);

[x,y,u,v] = openpiv( im1, im2, ...
    nx, ny, ...
    overlap_px, overlap_py);

% not sure it's needed
u(isnan(u)) = 0;
v(isnan(v)) = 0;
u = medfilt2(u, [3 3]);  % size of the window
v = medfilt2(v, [3 3]);


Displ.x = x;
Displ.y = y; 
Displ.u = u;
Displ.v = v; 

File: bos_correlation_openpiv.py
import numpy as np
from openpiv_python import openpiv

def BOS_correlation_OpenPIV(im1, im2, nx_pixel, ny_pixel, overlap_x, overlap_y=None):
    """
    Perform PIV correlation between two images using OpenPIV
    
    Parameters:
    -----------
    im1 : ndarray
        First image
    im2 : ndarray
        Second image
    nx_pixel : int
        Width of the interrogation window
    ny_pixel : int
        Height of the interrogation window
    overlap_x : float
        Overlap ratio in x direction (0-1)
    overlap_y : float, optional
        Overlap ratio in y direction (0-1). If None, uses overlap_x.
        
    Returns:
    --------
    result : dict
        Dictionary containing displacement field data (x, y, u, v)
    """
    # If overlap_y is not provided, use overlap_x
    if overlap_y is None:
        overlap_y = overlap_x
    
    # Calculate overlap in pixels
    ovlapHor = int(nx_pixel * (1 - overlap_x))
    ovlapVer = int(ny_pixel * (1 - overlap_y))
    
    # Call OpenPIV function
    x, y, u, v = openpiv(im1, im2, nx_pixel, ny_pixel, ovlapHor, ovlapVer)
    
    # Create result dictionary
    result = {
        'x': x,
        'y': y,
        'u': u,
        'v': v
    }
    
    return result
